src/data_write.py

```diff
@@ -996,238 +766,171 @@ def find_expectation_value(x, parameters, field_name):
+                                                      dtype=np.uint32)
 
-            for slice_id, parameters in parameters_holder.items():
-                name = dataset_name.format(sliceid=slice_id, dformat="bfiq")
+            for slice_num, slice_data in aveperiod_data.items():
+                name = dataset_name.format(sliceid=slice_num, dformat="bfiq")
                 output_file = dataset_location.format(name=name)
+                two_hr_file_with_type = self.slice_filenames[slice_num].format(ext="bfiq")
 
-                two_hr_file_with_type = self.slice_filenames[slice_id].format(ext="bfiq")
-
-                write_file(output_file, parameters, two_hr_file_with_type)
+                write_file(output_file, slice_data, two_hr_file_with_type, "bfiq")
 
-        def write_antenna_iq_params(parameters_holder):
+        def write_antenna_iq_params(aveperiod_data):
             """
             Writes out any pre-beamformed IQ that has been parsed. Adds additional slice info
             to each paramater dict. Some variables are captured from outer scope. Pre-beamformed iq
             is the individual antenna received data. Antenna_arrays_order will list the antennas' order.
 
-            :param  parameters_holder:  A dict that hold dicts of parameters for each slice.
-            :type   parameters_holder:  dict
+            :param  aveperiod_data:  Dict that holds SequenceData for each slice.
+            :type   aveperiod_data:  dict
             """
 
-            needed_fields = [
-            "borealis_git_hash",    "experiment_id",                "experiment_name",
-            "experiment_comment",   "num_slices",                   "slice_comment",
-            "station",              "num_sequences",                "rx_sample_rate",
-            "scan_start_marker",    "int_time",                     "tx_pulse_len",
-            "tau_spacing",          "main_antenna_count",           "intf_antenna_count",
-            "freq",                 "samples_data_type",            "pulses",
-            "sqn_timestamps",       "beam_nums",                    "beam_azms",
-            "data_dimensions",      "data_descriptors",             "antenna_arrays_order",
-            "data",                 "num_samps",                    "pulse_phase_offset",
-            "noise_at_freq",        "data_normalization_factor",    "blanked_samples",
-            "slice_id",             "slice_interfacing",            "scheduling_mode",
-            "gps_locked",           "gps_to_system_time_diff",      "agc_status_word",
-            "lp_status_word"
-            ]
-
             antenna_iq = data_parsing.antenna_iq_accumulator
             slice_id_list = [x for x in antenna_iq.keys() if isinstance(x, int)]
 
-            # Pop these so we don't include them in later iteration.
-            data_descriptors = antenna_iq.pop('data_descriptors', None)
```

Comment:
Why do we not need to pop here but we did with bfiq? Confused.

Reply:
I have no idea why the pop's were there in the first place. I removed this one, but missed the other one in write_bfiq(). I have removed it now as well.